lib/sycamore/sycamore/query/operators/query_database.py

```diff
@@ -42,3 +42,54 @@ class QueryDatabase(LogicalOperator):
     Whenever possible, use the query parameter to filter data at the source, as this is more
     efficient than filtering data in subsequent data filtering operators.
     """
+
+
+class QueryVectorDatabase(LogicalOperator):
+    """OpenSearch knn query to load records that are similar to a given query string. Returns the top k records
```

Comment:
I think it's best if the planner does not need to know about OpenSearch or implementation details. That is, from the planner's perspective, this is just a "SimilaritySearch" operation --maybe you don't even want to call it "QueryVectorDatabase". I would not say things about "knn" or "cosine similarity" since the planner need not understand those details.

Reply:
I'll remove knn/cosine but need opensearch in there since we're still using it to generate filters in opensearch dsl

Reply:
Or will remove as much of it as I can while maintaining filter support